_src/_posts/2018-09-07-writing-custom-presto-functions.markdown

```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title:  Overview of Jampp's platform
+date:   
+tag:    technology
+categories:
+    - infrastructure
+    - technology
+keywords: "Presto, PrestoDB, UDF, aggregation"
+author: dantepawlow
+---
+
+# Writing Custom PrestoDB Functions
+
+<!--excerpt.start-->
+Here at Jampp we process and analyse large amounts of data. One of the tools we employ to do so is [PrestoDB](https://prestodb.io/), which is a "Distributed SQL Query Engine for Big Data".
+Presto comes with many [functions](https://prestodb.io/docs/current/functions.html), which are usually enough for most use cases. Nevertheless, sometimes you need to implement your own function for a very specific use.
+
+Enter the [User Defined Functions](https://prestodb.io/docs/current/develop/functions.html) (UDFs, for short). Writing one is not as straightforward as it first appears, mainly because the information to do so is very scattered around the web (and across many Presto versions).
+
+In this blogpost, we present our _JSON Sum_ function, how did we write it, and some of the lessons we learned along the way.
+<!--excerpt.end-->
+
+## Function types
+
+Presto has two main types of functions: scalar and aggregation[^1]. Scalar functions behave more like traditional functions, in the sense that their output is immediate from the values of their parameters. On the other hand, aggregation functions take multiple rows as input and combine them into a single output, we'll focus mainly on these functions, as they're more complex (and, interesting!).
```

Comment:
`... a single output, we'll focus mainly on these functions, as they're more complex (and, interesting!).`

I would rephrase it as:

`... a single output. We'll focus mainly on the latter, as they're more complex (and, interesting!).`

Reply:
Good suggestion!